### DISCUSSION TITLE
SelectOneListBox breakes flow if used in a dynamic dialog

### DISCUSSION THREAD

**Discussion by wnibes** [primefaces/community#493] [PrimeFaces] [answered]
Hello together!
(Conditions are: PrimeFaces 13.0.3, Wildfly 29.0.1, Java 17)
I try to use a SelectOneListBox in a dynamic dialog started with
```
DialogFrameworkOptions options = DialogFrameworkOptions.builder()...
PrimeFaces.current()
            .dialog()
            .openDynamic("/dialogs/editCommission", options, null);
```
The dialog appears correctly and the data provieded by the backing bean is also shown well.
The dialog provides 2 CommandButtons (Cancel, Save) that have corresponding methods in the backing bean.
So far all seams quite well.
Case 1:
I did not select or change an item in the SelectOneListBox; both CommandButtons work well.
Case 2:
I change the selected item of the SelectOneListBox by clicking; both CommandButtons are dead (the close cross (upper right corner) of the dialog is still working.

Somebody out there who experienced the same?

**Comment by melloware** [comment, marked as answer]
Without seeing your code or knowing anything about it most likely you are getting some Form Validation error back and just not displaying it?  Both buttons "appear" dead but are probably submitting and getting an error back. You should be able to look in F12 Network traffic to confirm if network requests are being sent when you click those dead buttons.

**Reply by wnibes** [reply]
Thank you for that detail.
And you are right; the description "dead" was not precise enougth. Parallel to the browsers analysis tools I have an AuthorizationFilter, that logs any activity and I can see, that the buttons are triggering requests in both cases. But in the second case, there is no reaction.
That seams to me like the server side isn't able to "understand" the request in the second case. So the question is: Why does the server side do not understand the request anymore after selecting an item in SelectOneListBox?

```
<h:body>
	<h:form id="editCommission">
		<p:panelGrid id="panelGridId" styleClass="Xui-noborder" columns="2" layout="grid">
			<p:row>
				<p:column>#{msg['datum']}</p:column>
				<p:column>#{viewStaffBookingsEditCommission.staffInfoBooking.datum}</p:column>
			</p:row>
			<p:row>
				<p:column>#{msg['commission']}</p:column>
				<p:column>
					<p:selectOneListbox  id="advanced" value="#{viewStaffBookingsEditCommission.idicsPzeCommission}" var="c" onchange="#{viewStaffBookingsEditCommission.onChange()}" >
						<f:selectItems value="#{viewStaffBookingsEditCommission.commissions}" var="comm" itemLabel="#{comm.name}" itemValue="#{comm}" />
						<p:column>
							<h:outputText value="#{c.name}" />
						</p:column>
					</p:selectOneListbox>
					</p:column>
				</p:row>
		</p:panelGrid>
		<p:card>
			<p:commandButton value="#{msg['abbrechen']}" action="#{viewStaffBookingsEditCommission.cancel()}"  />
			&#160;
			<p:commandButton value="#{msg['speichern']}" action="#{viewStaffBookingsEditCommission.safeCommission()}" />
		</p:card>
	</h:form>
</h:body>
```
The xhtml page is also very simple. As I said, it is a dynamic dialogue.

**Reply by melloware** [reply]
Yep now look at the HTTP REsponse of those dead requests and look for `validationFailed`

**Reply by wnibes** [reply]
You are so right!
`<partial-response><changes><update id="j_id1:jakarta.faces.ViewState:0"><![CDATA[-3849149947142937854:-2674137915749100486]]></update><extension ln="primefaces" type="args">{&#34;validationFailed&#34;:true}</extension></changes></partial-response>`
But now I'm more confused: I do not have any validation (and there are no indicators like exceptions or similar, that show me that something went wrong). What do I miss?

**Reply by melloware** [reply]
Nope it means in your `<h:form>` you have something being submitted that is failing validation like required="true" or something.  Not my first rodeo so i knew this was your issue. Add a Messages component it will spit out what the error is.

**Reply by wnibes** [reply]
And once more you are so right!
Message tells me that converter is null. So far so clear the cause of the problem.
Implemented a Converter<T> and Voila! all is fine. So I have to thank you again ;-)

BUT: 
I'm facing two behaviours:

If I use the converter property of SelectOneListBox
`converter="MyConverter"`
it is working without any problems.

If I do not use this parameter, the converter is also instantiated, but with missing injection (log==null); so an exception is thrown.

Converter is defined as:

```
@Named
@ApplicationScoped
@FacesConverter(forClass = IdicsPzeCommission.class)
public class MyConverter implements Converter<IdicsPzeCommission> {

    @Inject
    Logger                              log;

    @Override
    public XXX getAsObject(FacesContext context, UIComponent component, String value) {
        log.info("bla");  --> java.lang.NullPointerException: Cannot invoke "org.slf4j.Logger.log..." because "this.log" is null
```

What is my fault on this?

**Reply by melloware** [reply]
Not sure of your JSF version or how you are producing your injected logger but FacesValidator in previous JSF did not support CDI Injection so you might want to check.

Also don't give it a scope of @ApplicationScoped see how the PF showcase does it...

```java
@Named
@FacesConverter(value = "countryConverter", managed = true)
public class CountryConverter implements Converter<Country> {

    @Inject
    private CountryService countryService;

    @Override
    public Country getAsObject(FacesContext context, UIComponent component, String value) {
        if (value != null && value.trim().length() > 0) {
            try {
                return countryService.getCountriesAsMap().get(Integer.parseInt(value));
            }
            catch (NumberFormatException e) {
                throw new ConverterException(new FacesMessage(FacesMessage.SEVERITY_ERROR, "Conversion Error", "Not a valid country."));
            }
        }
        else {
            return null;
        }
    }

    @Override
    public String getAsString(FacesContext context, UIComponent component, Country value) {
        if (value != null) {
            return String.valueOf(value.getId());
        }
        else {
            return null;
        }
    }
}
```

**Reply by wnibes** [reply]
That's cool. But it was not part of the showcase...

**Reply by melloware** [reply]
Its right here: https://www.primefaces.org/showcase/ui/input/autoComplete.xhtml?jfwid=b3894

**Reply by wnibes** [reply]
Right! But how should someone know if looking on "SelectOneListBox"? A link would be very helpful... :-))
